src/game/room.c

```diff
@@ -752,68 +730,65 @@ static void Room_PopulateSectorData(
     } while (!(fd_entry & END_BIT));
 }
 
-void Room_TestTriggers(int16_t *data, bool heavy)
+void Room_TestTriggers(const ITEM_INFO *const item)
 {
-    if (!data) {
-        return;
-    }
+    const bool is_heavy = item->object_number != O_LARA;
+    int16_t room_num = item->room_number;
+    const SECTOR_INFO *const sector =
+        Room_GetSector(item->pos.x, MAX_HEIGHT, item->pos.z, &room_num);
 
-    if ((*data & DATA_TYPE) == FT_LAVA) {
-        if (!heavy && Lava_TestFloor(g_LaraItem)) {
-            Lava_Burn(g_LaraItem);
-        }
-
-        if (*data & END_BIT) {
-            return;
-        }
-
-        data++;
+    if (!is_heavy && sector->is_death_sector && Lava_TestFloor(item)) {
+        Lava_Burn(g_LaraItem);
     }
 
-    int16_t type = (*data++ >> 8) & 0x3F;
-    int32_t switch_off = 0;
-    int32_t flip = 0;
-    int32_t new_effect = -1;
-    int16_t flags = *data++;
-    int16_t timer = flags & 0xFF;
+    const TRIGGER *const trigger = sector->trigger;
+    if (trigger == NULL) {
+        return;
+    }
 
     if (g_Camera.type != CAM_HEAVY) {
-        Camera_RefreshFromTrigger(type, data);
+        Camera_RefreshFromTrigger(trigger);
     }
 
-    if (heavy) {
-        if (type != TT_HEAVY) {
+    bool switch_off = false;
+    bool flip_map = false;
+    int32_t new_effect = -1;
+    ITEM_INFO *camera_item = NULL;
+
+    if (is_heavy) {
+        if (trigger->type != TT_HEAVY) {
             return;
         }
     } else {
-        switch (type) {
+        switch (trigger->type) {
+        case TT_TRIGGER:
+            break;
+
         case TT_SWITCH: {
-            int16_t value = *data++ & VALUE_BITS;
-            if (!Switch_Trigger(value, timer)) {
+            if (!Switch_Trigger(trigger->item_index, trigger->timer)) {
                 return;
             }
-            switch_off = g_Items[value].current_anim_state == LS_RUN;
+            switch_off =
+                g_Items[trigger->item_index].current_anim_state == LS_RUN;
             break;
         }
 
         case TT_PAD:
         case TT_ANTIPAD:
-            if (g_LaraItem->pos.y != g_LaraItem->floor) {
+            if (item->pos.y != item->floor) {
```

Comment:
Could this have an implication where an item other than Lara could trigger an antipad?

Reply:
That'd make for some interesting skips for sure 😄 
Other items will enter the `is_heavy` condition above (line 758) so it's only ever Lara in this `else` part 😌 I just thought a couple of references fewer to a global was nice where possible.

Reply:
Oh nice. I recently saw a playthrough of a custom level and the person tried to trigger a floor pad with a dead animal. Kinda genius but it didn't work. :laughing: 

So could we change `Lava_Burn(g_LaraItem);` to `Lava_Burn(item);` just to get rid of the global in that whole function?

Reply:
Oh yes. Although I think we'll have to cast to non const.